frontend/gradle/libs.versions.toml

```diff
@@ -0,0 +1,40 @@
+[versions]
+activityCompose = "1.9.3"
+# @pin incompatible otherwise
+agp = "8.6.0"
+composeBom = "2024.10.00"
+coreKtx = "1.13.1"
+espressoCore = "3.6.1"
+junit = "4.13.2"
+junitVersion = "1.2.1"
+kotlin = "2.0.21"
+lifecycleRuntimeKtx = "2.8.6"
+koin = "4.0.0"
+
+[libraries]
+androidx-activity-compose = { module = "androidx.activity:activity-compose", version.ref = "activityCompose" }
+androidx-compose-bom = { module = "androidx.compose:compose-bom", version.ref = "composeBom" }
+androidx-core-ktx = { module = "androidx.core:core-ktx", version.ref = "coreKtx" }
+androidx-espresso-core = { module = "androidx.test.espresso:espresso-core", version.ref = "espressoCore" }
+androidx-junit = { module = "androidx.test.ext:junit", version.ref = "junitVersion" }
+androidx-lifecycle-runtime-ktx = { module = "androidx.lifecycle:lifecycle-runtime-ktx", version.ref = "lifecycleRuntimeKtx" }
+androidx-material3 = { module = "androidx.compose.material3:material3" }
+androidx-ui = { module = "androidx.compose.ui:ui" }
+androidx-ui-graphics = { module = "androidx.compose.ui:ui-graphics" }
+androidx-ui-test-junit4 = { module = "androidx.compose.ui:ui-test-junit4" }
+androidx-ui-test-manifest = { module = "androidx.compose.ui:ui-test-manifest" }
+androidx-ui-tooling = { module = "androidx.compose.ui:ui-tooling" }
+androidx-ui-tooling-preview = { module = "androidx.compose.ui:ui-tooling-preview" }
+junit = { module = "junit:junit", version.ref = "junit" }
+koin-core = { module = "io.insert-koin:koin-core", version.ref="koin"}
+koin-android = { module = "io.insert-koin:koin-android", version.ref="koin" }
+koin-test = { module= "io.insert-koin:koin-test-junit4", version.ref="koin" }
+
+[plugins]
+android-application = { id = "com.android.application", version.ref = "agp" }
+com-github-benmaes-versions = "com.github.ben-manes.versions:0.51.0"
+compose-compiler = { id = "org.jetbrains.kotlin.plugin.compose", version.ref = "kotlin" }
+kotlin-android = { id = "org.jetbrains.kotlin.android", version.ref = "kotlin" }
+nl-littlerobots-versioncatalogueupdate = "nl.littlerobots.version-catalog-update:0.8.4"
+org-cyclonedx-bom = {id="org.cyclonedx.bom" , version="1.10.0"}
+com-ncorti-ktfmt-gradle = { id = "com.ncorti.ktfmt.gradle", version="0.20.1"}
```

Comment:
move the versions up like for all other dependencies so it is consistent with the:

```
{ id = "...", version.ref = "" }
```
syntax.

Reply:
also run the `versionCatalogUpdate` tasks once, because it reorders the entries in the file quite a bit, which leads to noisy commits in the future.

Reply:
Done! I also added a combinedFormat task that will run ktfmtFormat and versionCatalogFormat, i think we should use this task when creating pre-commit/pre-push hooks.